MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift

```diff
@@ -7,16 +7,20 @@
 
 import Foundation
 
-class MovieDetailsViewModel {
+protocol MovieDetailsViewModelType {
+    func fetchMovieDetails()
+}
+protocol MovieDetailsView: AnyObject {
+    func updateUI()
+}
+
+class MovieDetailsViewModel: MovieDetailsViewModelType {
 
     // MARK: - Variables
     private let movieDetailsRepository: MovieDetailsRepositoryType
     private var movieID = 0
-    var movieDetails: MovieDetails? {
-        didSet {
-            didUpdateDetails?()
-        }
-    }
+    var movieDetails: MovieDetails?
+    weak var view: MovieDetailsView?
```

Comment:
this is very very very strange. The viewModel should never ever be coupled to the view. Please change this. This can't be approved. It breaks architecture in every regard.
The view knows about the viewModel, not the other way round

Reply:
will address